src/reader.cpp

```diff
@@ -25,24 +25,32 @@ using namespace LAMMPS_NS;
 Reader::Reader(LAMMPS *lmp) : Pointers(lmp)
 {
   fp = nullptr;
+  binary = false;
+  compressed = false;
 }
 
 /* ----------------------------------------------------------------------
    try to open given file
-   generic version for ASCII files that may be compressed
+   generic version for ASCII files that may be compressed or native binary dumps
 ------------------------------------------------------------------------- */
 
 void Reader::open_file(const std::string &file)
 {
   if (fp != nullptr) close_file();
 
   if (platform::has_compress_extension(file)) {
-    compressed = 1;
+    compressed = true;
     fp = platform::compressed_read(file);
     if (!fp) error->one(FLERR,"Cannot open compressed file for reading");
```

Comment:
Should we add error info for not supporting the compressed binary?

Reply:
if we change things here, we need to make the corresponding changes to the dump styles. Right now, compressed binary files are not detected nor generated; if a file ends in .bin.gz it will be the same as .gz. For reading, we should have the same behavior. If somebody externally compresses a binary file and tries to read that, it will create the same error as if you would rename a binary file to not end in .bin, i.e. incorrect format.